core/templates/dev/head/pages/exploration_editor/translation_tab/TranslatorOverviewDirective.js

```diff
@@ -30,24 +30,65 @@ oppia.directive('translatorOverview', [
         '/pages/exploration_editor/translation_tab/' +
         'translator_overview_directive.html'),
       controller: [
-        '$scope', '$rootScope', '$window', 'SUPPORTED_AUDIO_LANGUAGES',
-        'LanguageUtilService', 'TranslationLanguageService',
-        'TranslationStatusService', 'DEFAULT_AUDIO_LANGUAGE', function(
-            $scope, $rootScope, $window, SUPPORTED_AUDIO_LANGUAGES,
-            LanguageUtilService, TranslationLanguageService,
-            TranslationStatusService, DEFAULT_AUDIO_LANGUAGE) {
+        '$rootScope', '$scope', '$timeout', '$window',
+        'ExplorationLanguageCodeService', 'LanguageUtilService',
+        'TranslationLanguageService', 'TranslationStatusService',
+        'TranslationTabActiveModeService', 'DEFAULT_AUDIO_LANGUAGE',
+        'SUPPORTED_AUDIO_LANGUAGES', function(
+            $rootScope, $scope, $timeout, $window,
+            ExplorationLanguageCodeService, LanguageUtilService,
+            TranslationLanguageService, TranslationStatusService,
+            TranslationTabActiveModeService, DEFAULT_AUDIO_LANGUAGE,
+            SUPPORTED_AUDIO_LANGUAGES) {
           var LAST_SELECTED_TRANSLATION_LANGUAGE = (
             'last_selected_translation_lang');
           var prevLanguageCode = $window.localStorage.getItem(
             LAST_SELECTED_TRANSLATION_LANGUAGE);
           var allAudioLanguageCodes = LanguageUtilService
             .getAllAudioLanguageCodes();
+
+          $scope.VOICEOVER_MODE = 'Record';
+          $scope.TRANSLATION_MODE = 'Translate';
+
+          TranslationTabActiveModeService.activateVoiceoverMode();
+          $scope.getStaticImageUrl = UrlInterpolationService.getStaticImageUrl;
           $scope.languageCode =
             allAudioLanguageCodes.indexOf(prevLanguageCode) !== -1 ?
               prevLanguageCode : DEFAULT_AUDIO_LANGUAGE;
           TranslationLanguageService.setActiveLanguageCode(
             $scope.languageCode);
-          $scope.languageCodesAndDescriptions = (
+          $scope.inTranslationMode = false;
+          $scope.inVoiceoverMode = true;
+          $scope.showTabModeSwitcher = function() {
+            return ($scope.languageCode !== (
+              ExplorationLanguageCodeService.displayed));
+          };
+
+          $scope.showModeSwitchHelperImage = function() {
```

Comment:
Why does this function return something? A show* function is not expected to return something I think.

Reply:
yeah can/should in front of the current name would be good.